src/Compilers/CSharp/Portable/Utilities/TypeSymbolExtensions.cs

```diff
@@ -119,7 +147,7 @@ public static bool HasCustomModifiers(this TypeSymbol type, bool flagNonDefaultA
 
                                 foreach (TypeWithAnnotations typeArg in typeArgs)
```

Comment:
>foreach (TypeWithAnnotations typeArg in typeArgs) [](start = 32, length = 49)

Consider using:
```
if (typeArgs.Any(
    (p, f) => checkTypeWithAnnotations(p, f),
    flagNonDefaultArraySizesOrLowerBounds))
```

Reply:
These methods can be called on very deeply nested generics. I'd prefer to avoid any kind of capturing.

---
In reply to: [422383386](https://github.com/dotnet/roslyn/pull/44078#discussion_r422383386) [](ancestors = 422383386)